[stayWefolio 서수연] Add: Nav layout 작성

suyeon-DV opened the pull request:
## 수정 사항 간략한 한줄 요약
[2020/10/23]
- Refactor: 멘토 리뷰 반영

[2020/10/21]
- Refactor: 반복되는 li 리팩토링

[2020/10/20]
- Add: Nav layout 작성
		 
## 수정 사항들 자세한 내용

[2020/10/23] Refactor: 멘토 리뷰 반영
- mock data 따로 분리
- mixin을 변수 이용에서 사용하도록 변경
- 스타일 적용되지 않는 nesting 삭제
- width, height 100% 모두 삭제 후 각 요소가 공간을 차지하도록 수정
- 공통된 스타일이 적용되는 className 삭제 후 하나의 Tag만을 사용하도록 수정
- className의 동적사용에 따라 스타일이 변경되도록 수정

[2020/10/21] Refactor: 반복되는 li 리팩토링
- 반복되는 li를 map 사용해서 리팩토링

[2020/10/20] Add: Nav layout 추가
- Nav layout 작성 완료
- Nav scss 작성 완료

## 기타 질문 및 특이 사항
(없음)

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

suyeon-DV committed:
Add: layout 작성
- 전반적인 layout 완성함

suyeon-DV committed:
Add: navbar layout 추가
- container layout 추가

suyeon-DV committed:
Remove: 주석 삭제
- 불필요한 주석 삭제

suyeon-DV committed:
Remove: 불필요한 폴더 삭제
- dataSSY폴더 내 data.json 파일 삭제

suyeon-DV committed:
Modify: JSX내 className 수정
- JSX내 class attribute를 className으로 수정

suyeon-DV committed:
Add: 미디어쿼리 추가
- 미디어 쿼리 추가
- 브랜드 로고 이미지 추가

suyeon-DV committed:
Modify: layout 수정
- menu layout 수정

claire-euni commented on src/components/Nav/Nav.scss:
```diff
@@ -1 +1,197 @@
\ No newline at end of file
+@import "../../styles/common.scss";
+
+* {
+  font-family: "Apple SD Gothic Neo", "Nanum Gothic", "나눔고딕",
+    "Malgun Gothic", "맑은고딕", Dotum, "돋움", Gulim, "굴림", "Helvetica Neue",
+    Helvetica, Arial, sans-serif;
+}
```
common.scss에서 설정하면 되지 않을까요? :)

suyeon-DV replied:
font-family를 common.scss에 공통 적용하는 걸 조원과 상의해보기로 했습니다!! 상의 후에 수정하겠습니다.

claire-euni commented on src/components/Nav/Nav.js:
```diff
@@ -1,8 +1,92 @@
+              </form>
+              <div className="mediaIconsWrap">
+                <ul className="mediaIcons">
+                  <li className="facebook">
+                    <a href="https://www.facebook.com/">
+                      <i className="fab fa-facebook-f"></i>
+                    </a>
+                  </li>
```
<li> 태그들이 반복되고 있는데 map()을 활용해보세요 🤗

suyeon-DV replied:
네 감사합니다. 이전 리팩토링에서 했던 내용들인데.... 당연하다는 듯이 또 들어갔습니다... 반성합니다...😭
코드 보시느라 고생 많으세요. 감사합니다. 전부 수정하겠습니다....ㅠㅠ.......

claire-euni commented on src/components/Nav/Nav.js:
```diff
@@ -1,8 +1,92 @@
+              </div>
+            </div>
+            <ul className="bottomLineWrap">
+              <li className="careby">
+                <span className="carebyLink">CARE BY</span>
+              </li>
```
이 부분도 map()을 활용해보세요~

suyeon-DV replied:
map활용해서 리팩토링 진행했습니다.😀😀

claire-euni commented on src/components/Nav/Nav.scss:
```diff
@@ -1 +1,197 @@
+  border: 1px solid $border-gray;
+}
+
+@media (min-width: 768px) {
+  .container {
+    width: 750px;
+  }
+}
+@media (min-width: 992px) {
+  .container {
+    width: 970px;
+  }
+}
+@media (min-width: 1200px) {
+  .container {
+    width: 1170px;
+  }
+}
```
와우 반응형 구현도 하시고👍👍

suyeon-DV replied:
감사합니다😁

claire-euni commented on src/components/Nav/Nav.scss:
```diff
@@ -1 +1,197 @@
+}
+
+.Nav {
+  width: 100%;
```
필요한 속성일까요~?

suyeon-DV replied:
삭제했습니다. 이것도 이전에 받은 리뷰에서 있었던 내용이네요..... ㅠㅠ 확인하겠습니다.

claire-euni commented on src/components/Nav/Nav.scss:
```diff
@@ -1 +1,197 @@
+          justify-content: space-around;
+
+          li {
+            // width: 20%;
```
주석은 삭제해주세요 🤟

suyeon-DV replied:
네 모든 주석 삭제했습니다.

claire-euni commented on src/index.js:
```diff
@@ -2,6 +2,7 @@ import React from "react";
 import { BrowserRouter } from "react-router-dom";
 import Routes from "./Routes";
+import "@fortawesome/fontawesome-free/js/all.js";
 
```
불필요한 엔터는 삭제해주세요 :)

suyeon-DV replied:
import 적은 곳에서 불필요한 엔터 삭제 했습니다.

claire-euni reviewed (commented):
첫 PR 축하드립니다 🌼

suyeon-DV committed:
Refactor: 반복되는 li 리팩토링
- 반복되는 li를 map 사용해서 리팩토링

saengmotmi commented on src/components/Nav/Nav.js:
```diff
@@ -1,8 +1,108 @@
+import "./Nav.scss";
 
-export class Nav extends Component {
+const brandLogos = [
```
mock data 내용이 많네요! 별도 파일로 분리해줄 수 있을 것 같습니다

suyeon-DV replied:
mock data로 파일 분리했습니다. 감사합니다😀😀

saengmotmi commented on src/components/Nav/Nav.scss:
```diff
@@ -1 +1,158 @@
+  left: 0;
+  right: 0;
+  z-index: 999;
+  @include flex-center;
```
flex 관련 속성들을 묶어서 재활용해준 건 좋은데, 이런 방식이라면 모든 경우에 대해 미리 설정을 해줘야 할 것 같네요. [링크](https://sass-lang.com/documentation/at-rules/mixin) 확인하셔서 Arguments 항목 한번 참고해보세요. 함수처럼 인자로 필요한 값들을 넘겨주는 방식을 적용해볼 수 있습니다.

suyeon-DV replied:
mixin에 인자를 설정할 수 있는지는 몰랐습니다. 처음 내용을 봤을 때 꽤나 허들이 있었지만.... 덕분에 정말 좋은 걸 알게되었습니다!!!😁😁 코드가 길면 길수록 큰 도움이 될 것같아요!!!! 앞으로 열심히 써서 익혀보겠습니다. 
일단 기존에 사용하던 mixin flex-center는 팀원이랑 같이 사용하던 부분이라 미팅 때 이야기 해보고 삭제하겠습니다. 감사합니다🥰🥰

saengmotmi commented on src/components/Nav/Nav.scss:
```diff
@@ -1 +1,158 @@
+              height: 100%;
+              @include flex-center;
+
+              a {
+                img {
+                }
+              }
```
불필요한 내용은 제거해주세요

suyeon-DV replied:
스타일 적용을 하지 않는 네스팅은 모두 제거 했습니다!!!

saengmotmi commented on src/components/Nav/Nav.scss:
```diff
@@ -1 +1,158 @@
+      }
+    }
+
+    .navContainer {
+      width: 100%;
+      height: 100%;
+      display: flex;
+      flex-direction: column;
```
- width, height 100% 주고 계신데 꼭 필요한 속성인지 확인 부탁드립니다
- 세로 정렬을 하는 목적이라면 div 태그가 기본적으로 block 속성이라는 점을 이용하면 좋을 것 같네요! 다른 요소들에도 이러한 내용이 많은 것 같아, 전반적으로 확인 부탁드립니다.

suyeon-DV replied:
말씀해주신대로 개개의 요소에 padding이나 margin을 넣어 요소 하나가 해당 공간을 모두 가질 수 있도록 수정했습니다. 그러면서 width랑 height가 100%로 되어 있던부분 고쳤습니다. footer 하면서도 100% 많이 쓴 것같은데 footer도 리팩토링하고 댓글 그쪽에 다시 달겠습니다. 
좋은 방법 알려주셔서 감사합니다😍😍

saengmotmi commented on src/components/Nav/Nav.scss:
```diff
@@ -1 +1,158 @@
+          height: 100%;
+          @include flex-center;
+
+          .careByLink,
+          .magazineLink,
+          .pickLink,
+          .promotionLink,
+          .preorderLink,
+          .bookingLink {
```
동일한 스타일링을 주고 있는데 꼭 다른 이름으로 써야 할까요?

suyeon-DV replied:
태그 이름으로 묶어서 스타일링 했습니다. 
그리고 질문 드렸을 때 last-child 쓸거라고 했었는데 last-child가 적용이 안되어서 last-of-type도 했다가 결국 지영 멘토님의 도움을 받아서 조건문을 사용해 classname을 변경하는 방향으로 수정했습니다😭

saengmotmi commented on src/components/Nav/Nav.js:
```diff
@@ -1,8 +1,108 @@
+              <form className="searchBarWrap">
+                <input className="searchBar" type="text" placeholder="Search" />
+                <button className="searchBarBtn">
+                  <i class="fas fa-search"></i>
```
```suggestion
                  <i class="fas fa-search" />
```
셀프 클로징 태그 적용해주세요

suyeon-DV replied:
이 아이콘이 fontawesome의 아이콘인데 셀프 클로징을 하니 컴파일에러가 나서 변경하지 못 했습니다😭

saengmotmi commented on src/components/Nav/Nav.js:
```diff
@@ -1,8 +1,108 @@
+              </div>
+            </div>
+            <ul className="bottomLineWrap">
+              {menus.map((menu) => {
```
menus, menu 변수명 좋네요! 👍

suyeon-DV replied:
감사합니다😀😀

saengmotmi reviewed (commented):
리뷰 확인해주세요!

suyeon-DV committed:
Refactor: 멘토 리뷰 반영
- mock data 따로 분리
- mixin을 변수 이용에서 사용하도록 변경
- 스타일 적용되지 않는 nesting 삭제
- width, height 100% 모두 삭제 후 각 요소가 공간을 차지하도록 수정
- 공통된 스타일이 적용되는 className 삭제 후 하나의 Tag만을 사용하도록 수정
- className의 동적사용에 따라 스타일이 변경되도록 수정

claire-euni commented on src/styles/common.scss:
```diff
@@ -7,14 +7,14 @@ $color-title-black: rgb(51, 51, 51);
-    @content;
-  }
+@mixin flex($justify-content: center, $align-item: center) {
+  display: flex;
+  justify-content: $justify-content;
+  align-items: $align-item;
```
바로 center;를 쓰지 않고 $로 설정해주신 이유가 있나요?

suyeon-DV replied:
원래 space-between을 사용 해야할 경우가 있었는데.... 제가 사용 후 삭제할 부분 확인하면서 삭제한 것 같습니다😭
전체적으로 다시 보고 PR을 다시 작성하던지 다른 부분에서 변경할 수 있는 부분이 있는지 확인해보겠습니다😦😥
리뷰 보시느라 고생하셨습니다!!! 😍
